src/components/StatsScoringDialog.vue

```diff
@@ -1,8 +1,17 @@
 <template>
   <v-dialog v-model="show">
     <!-- Activator -->
-    <template #activator="{ props }">
-      <v-btn color="primary" class="mb-2" v-bind="props" variant="plain">
+    <template #activator="{ attrs, props }">
+      <v-btn
+        color="primary"
+        class="mb-2"
+        v-bind="{
+          ...$props,
+          ...props,
+          ...$attrs,
+          ...attrs,
```

Comment:
what's going on here? Where do `$props` and `$attrs` come from?

Reply:
I know this is super weird, and eventually I think we should rework how we do this activator, but let me explain. This entirely has to do with how we're applying the data attribute to the component calling context.

In `src/components/CreateGameDialog.vue` we call `<stats-scoring-dialog :show-button-text="false" data-cy="ranked-info-button" />` which has the data attr we need.

Then in `src/components/StatsScoringDialog.vue` we have an activator defined (the output of that component's dialog activator) which is what needs that data attr. The activator in this component looks like this.
```html
<!-- Activator -->
<template #activator="{ attrs, props }">
  <v-btn
    color="primary"
    class="mb-2"
    v-bind="{
      ...$props,
      ...props,
      ...$attrs,
      ...attrs,
    }"
    variant="plain">
    <span v-if="showButtonText">How are ranks determined?</span>
    <v-icon class="ml-1" icon="mdi-information-outline" />
  </v-btn>
</template>
```

The issue is that the `v-bind` was only applying the props from a defined activator but we aren't defining one. We have two options, we can duplicate the button and do something like this.
```html
<v-btn color="primary" class="mb-2" variant="plain" data-cy="ranked-info-button">
  <v-icon class="ml-1" icon="mdi-information-outline" />
</v-btn>
```

...or we can proxy the attributes and props down from the parent on to the default activator via `$props` and `$attrs`, which are the parent values (that you saw on the `stats-scoring-dialog` component above). This would allow any calling context to apply attrs and props to the default activator element by binding them along with the activator props as far as I understand.

That said, I pushed up the other example that is a bit less DRY, but easier to understand.

See the following for more information, https://vuejs.org/api/component-instance.html#attrs.